src/Server/Query.elm

```diff
@@ -149,13 +149,79 @@ ingredientParser { countries, ingredients } string =
                     )
                 |> RE.andMap (Ok Nothing)
 
+        [ id, mass, variant, countryCode, byPlane, bonuses ] ->
+            let
+                ingredient =
+                    ingredients
+                        |> Ingredient.findByID (Ingredient.idFromString id)
+            in
+            Ok BuilderQuery.IngredientQuery
+                |> RE.andMap (Result.map .id ingredient)
+                |> RE.andMap (validateMass mass)
+                |> RE.andMap (variantParser variant)
+                |> RE.andMap (foodCountryParser countries countryCode)
+                |> RE.andMap
+                    (ingredient
+                        |> Result.andThen
+                            (\ingredientResult ->
+                                validateByPlaneValue byPlane ingredientResult
+                                    |> Result.andThen
+                                        (\maybeByPlane ->
+                                            Ingredient.byPlaneAllowed maybeByPlane ingredientResult
+                                        )
+                            )
+                    )
```

Comment:
Maybe this could be extracted to a helper like `byPlaneParser` to be reused between the last two case branches?